src/app-layout/visual-refresh/context.tsx

```diff
@@ -394,68 +396,58 @@ export const AppLayoutInternalsProvider = React.forwardRef(
       [props.onSplitPanelPreferencesChange, setSplitPanelPreferences, setSplitPanelLastInteraction]
     );
 
-    /**
-     * The activeDrawerWidth is required in JavaScript to acccurately calculate whether a SplitPanel
-     * in the side position should be forced to the bottom based on available horiziontal space.
-     *
-     * The handleDrawersClick will either open a new drawer or close the currently open drawer by setting
-     * the activeDrawerId value. The active drawer can also be closed if a user clicks the Tools trigger
-     * button. This will skip the focus handling because the focus should be going to the Tools close
-     * button and not one of the drawers trigger buttons.
-     *
-     * The drawersTriggerCount is computed in order to determine whether the triggers should be persistent
-     * in the UI when a drawer is open. The trigger button container is suppressed when a drawer is open
-     * and their is only one trigger button.
-     *
-     * The hasDrawerViewportOverlay property is used to determine if any drawer is obscuring the entire
-     * viewport. This currently applies to Navigation, Tools, and Drawers in mobile viewports.
-     */
-    const drawers = (props as InternalDrawerProps).drawers;
-
-    const [activeDrawerId, setActiveDrawerId] = useControllable(drawers?.activeDrawerId, drawers?.onChange, null, {
-      componentName: 'AppLayout',
-      controlledProp: 'drawers.activeDrawerId',
-      changeHandler: 'onChange',
-    });
+    const {
+      drawers,
+      activeDrawer,
+      activeDrawerId,
+      onActiveDrawerChange,
+      onActiveDrawerResize,
+      activeDrawerSize,
+      ...drawersProps
+    } = useDrawers(
+      props as InternalDrawerProps,
```

Comment:
I'm a bit confused with this line, so from my understanding when we render the AppLayout the runtime API attaches the required props, so we cast it here ( even if the props definition does not include `drawers` or `__disableRuntimeDrawers`.

should we then add some kind of guarding conditions in the `useDrawers` hook to check if the props have the required properties passed to it?

Reply:
Both of these properties are optional, what kind of guarding condition do see possible here?

Reply:
I see, I looked again and its fine.

It just does not look correct to me when we are casting the whole props as `InternalDrawerProps`, so we are sending a lot of un-required fields to the hook. as a minor thingy, can we selectively send them?

like:
```
const useDrawersFields = {
drawers: (props as InternalDrawerProps)?.drawers,
__disableRuntimeDrawers: (props as InternalDrawerProps)?. __disableRuntimeDrawers
}
```

Will approve tho, just a suggestion .

Reply:
I see. I will revisit it when `props.drawers` becomes public and will not require type casting